src/main.cpp

```diff
@@ -240,26 +240,38 @@ int main(int argc, const char **argv) {
 		else {
 			FILE* file = fopen(outputPath.generic_string().c_str(), "wb");
 			if (file != NULL) {
-				uint32_t* writeHeader = new uint32_t[2];
 				fwrite("glTF", sizeof(char), 4, file); // magic
+
+				uint32_t* writeHeader = new uint32_t[2];
 				writeHeader[0] = 2; // version
-				int padding = (20 + jsonString.length()) % 4;
-				if (padding != 0) {
-					padding = 4 - padding;
-				}
-				writeHeader[1] = 20 + jsonString.length() + padding + 8 + buffer->byteLength; // length
-				fwrite(writeHeader, sizeof(uint32_t), 2, file);
-				writeHeader[0] = jsonString.length() + padding; // chunkLength
+
+				const int HEADER_LENGTH = 12;
+				const int CHUNK_HEADER_LENGTH = 8;
+
+				int jsonPadding = (4 - (jsonString.length() & 3)) & 3;
```

Comment:
Ah, sorry I missed that case.

I thought about it a little bit, and I know modulo is more intuitive for me when reading, but that may not be true for everyone, so I'll leave that up to your discretion. Otherwise this is fine.